include/boost/json/detail/impl/number.ipp

```diff
@@ -92,9 +92,16 @@ pow10(int exp) noexcept
 
         1e+300, 1e+301, 1e+302, 1e+303, 1e+304, 1e+305, 1e+306, 1e+307, 1e+308 };
 
-    exp += 308;
-    BOOST_ASSERT(exp >= 0 && exp < 618);
-    return tab[exp];
+    if (exp < -308 || exp > 308)
+    {
+        return std::pow(10.0, exp);
```

Comment:
idk about this

Reply:
I'll get to it. I wanted to prevent the assert from happening

Reply:
Do we still need this check?

Reply:
we could just check for "more than 3 digits" and then rely on the finalize to detect the over/underflow. I think i'd prefer that...

oh wait. What if the input is 0.[ten thousand zeroes]1e10000 ?
there's no reason we shouldn't parse this correctly.

It would be trivial to compute the exponent limits on entry into this state

Reply:
In my view the check is superfluous. it's more expensive to perform the check than to fail late.

Reply:
In general I think it is good to fail as soon as we know we have an invalid input. So if we get 309 as the exponent, we can fail right there. I'm not happy about having a potential call to `std::pow`, especially when we know that the exponent is out of range.

Reply:
0.0000000001e309 is actually 1e289. stod will parse this correctly.